src/APK/Podman.hs

```diff
@@ -0,0 +1,186 @@
+{-# LANGUAGE CPP #-}
+
+module APK.Podman
+  ( Config(..)
+  , HostPaths(..)
+  , PackageName(..)
+  , build
+  , getHostPaths
+  , index
+  , run
+  , update
+  )
+where
+
+import APK.Prelude
+import Control.Monad (unless)
+import Data.Text.Encoding (decodeUtf8, encodeUtf8)
+import System.Exit (ExitCode(ExitSuccess))
+import System.Path ((</>))
+
+import qualified APK.TH                as TH
+import qualified Data.ByteString.Lazy  as LBS
+import qualified Data.List             as List
+import qualified Network.AWS           as AWS
+import qualified Network.AWS.Data.Body as AWS
+import qualified System.Path           as Path
+import qualified System.Path.Directory as Path
+import qualified System.Process.Typed  as Process
+
+newtype ContainerName = ContainerName Text
+  deriving newtype ToText
+
+newtype ImageName = ImageName Text
+  deriving newtype ToText
+
+newtype PackageName = PackageName Text
+  deriving newtype ToText
+
+data Config = Config
+  { arguments     :: [String]
+  , command       :: String
+  , repositoryPath :: Path.RelDir
+  , workPath       :: Path.AbsDir
+  }
+
+data HostPaths = HostPaths
+  { hostAbuildPath             :: Path.AbsDir
+  , hostCachePath              :: Path.AbsDir
+  , hostHomePath               :: Path.AbsDir
+  , hostPackagesPath           :: Path.AbsDir
+  , hostRepositoryPackagesPath :: Path.AbsDir
+  , hostRepositoryPath         :: Path.AbsDir
+  , hostWorkingPath            :: Path.AbsDir
+  }
+
+defaultConfig :: Config
+defaultConfig = Config
+  { arguments      = []
```

Comment:
Instance of https://github.com/mbj/apk-build/pull/1/commits/820343cc510bcea80ee040f1a890e2a93ef2dd8a#r387981992